adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -121,6 +125,102 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
     replaceRdd(tFn(rdd))
   }
 
+  private[rdd] def sorted: Boolean
+  private[rdd] def partitionMap: Option[Seq[Option[(ReferenceRegion, ReferenceRegion)]]]
+
+  /**
+   * This repartition method repartitions all data in this.rdd and distributes it as evenly as possible
+   * into the number of partitions provided. The data in this.rdd is already sorted (by ReferenceRegion.start)
+   * so we ensure that it is kept that way as we move data around.
+   *
+   * Data is repartitioned in lists to maintain the sorted order. After the repartition, the lists are placed
+   * in the correct order based on their original sorted order.
+   *
+   * @param partitions the number of partitions to repartition this.rdd into
+   * @return a new SortedGenomicRDDMixIn with the RDD partitioned evenly
+   */
+  private def evenlyRepartition(partitions: Int)(implicit tTag: ClassTag[T]): GenomicRDD[T, U] = {
+    require(sorted, "Cannot evenly repartition an unsorted RDD")
+    // the average number of records on each node will help us evenly repartition
+    val average: Double = rdd.count.toDouble / partitions
+    // we already have a sorted rdd, so let's just move the data
+    // but we want to move it in blocks to maintain order
+    // zipWithIndex seems to be the cheapest way to guarantee this
+    val finalPartitionedRDD = rdd
+      .mapPartitionsWithIndex((idx, iter) => {
+        // here we get the destination partition number
+        // this will calculate so that each partition has a nearly
+        // equal amount of data on it
+        extractReferenceRegionsForPartition(iter, partitionMap.get(idx))
+      }, preservesPartitioning = true).zipWithIndex
+      .mapPartitionsWithIndex((idx, iter) => {
+        // divide the global index by the average to get the destination
+        // partition number
+        iter.map(_.swap).map(f => ((f._2._1, (f._1 / average).toInt), f._2._2))
+      })
+      .repartitionAndSortWithinPartitions(new ReferenceRegionRangePartitioner(partitions))
+
+    val finalPartitionMap = finalPartitionedRDD.mapPartitions(iter =>
```

Comment:
You're going to want to persist `finalPartitionedRDD` before you run this, else you're going to trigger a lineage recompute.